### PR TITLE
Alerting: Add support for documenting which alerting APIs are stable

### PR DESCRIPTION
**Pull request by alexweav**
**What this PR does / why we need it**:

Adds support for separately documenting APIs that we consider to be stable. This lets us communicate to users much easier as to which APIs they can depend on.

Simply add `stable` to the route tags to mark a route as stable:
2022-05-16-103143_749x73_scrot

It will then appear in a separate OpenAPI doc that only contains the stable routes:
2022-05-16-102100_1265x615_scrot

We can then share links and point users to this new document, without changing the existing messaging that `post.json` is not considered stable.

Doesn't affect the existing doc or code generation - simply adds this new document on top.

**Which issue(s) this PR fixes**:

Rel #36153 
Rel https://github.com/grafana/grafana/issues/48100

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/api/tooling/Makefile [hunk 1]
```diff
@@ -11,24 +11,30 @@ PATH_UP = ../../../../..
...
 	@hash swagger &>/dev/null || (brew tap go-swagger/go-swagger && brew install go-swagger)
 
 spec.json-mac: ensure_go-swagger_mac $(GO_PKG_FILES)
-	swagger generate spec -m -w $(API_DIR) -o spec.json
+	swagger generate spec -m -w $(API_DIR) -o spec.json && swagger generate spec -m --include-tag=stable -o ngalert.json
 
 post.json: spec.json
 	go run cmd/clean-swagger/main.go -if $(<) -of $@
 
+api.json: spec-stable.json
```
Comment (alexweav) [inline pkg/services/ngalert/api/tooling/Makefile:35]: New file is called `api.json` which I thought was legacy/ng agnostic and official sounding enough. Open to suggestions here.

File: pkg/services/ngalert/api/tooling/cmd/clean-swagger/main.go [hunk 2]
```diff
@@ -63,6 +63,39 @@ func main() {
...
+				continue
+			}
+
+			tags, ok := op["tags"].([]interface{})
+			if !ok {
+				log.Println("invalid op, skipping")
+				continue
+			}
+
+			// Remove "stable" tag. Multiple tags cause routes to render strangely in the final docs.
```
Comment (alexweav) [inline pkg/services/ngalert/api/tooling/cmd/clean-swagger/main.go:89]: Hide the stable tag. It's just used for knowing which doc to send the route to. We don't want to codegen any extra go files based on it. Also, the swagger UI looks super weird when a route has multiple tags - the route shows up many times in the doc.

File: pkg/services/ngalert/api/tooling/post.json [hunk 3]
```diff
@@ -3114,6 +3114,7 @@
    "x-go-package": "github.com/prometheus/alertmanager/api/v2/models"
   },
   "alertGroup": {
+   "description": "AlertGroup alert group",
```
Comment (alexweav) [inline pkg/services/ngalert/api/tooling/post.json:3117]: Reviewers can skip changes in `post.json` and `spec.json` files. We thrash a few keys every time this file gets regenerated. This was not introduced by this PR. There are no other changes to these files.

**Review by papagian — APPROVED** [review]
The root Makefile [used to reference the `post.json`](https://github.com/grafana/grafana/blob/84d4ce51addb5f88abd53b9b2787d4c18409fc01/Makefile#L38); I think now it should point to `api.json`.
note: that `post.json` is referenced several times (unfortunately) in the root Makefile and therefore has to be replaced everywhere.

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Pull request merged**